app/services/action_builder.rb

```diff
@@ -1,11 +1,23 @@
+require_relative 'action_queue'
+
 module ActionBuilder
+  def build_action(extra_attrs = {})
+    subscribed_member = !existing_member?
 
-  def build_action
-    Action.create(
+    action = Action.create({
       member: member,
       page: page,
-      form_data: @params
-    )
+      form_data: @params,
+      subscribed_member: subscribed_member
+    }.merge(extra_attrs))
+
+
+    if action
```

Comment:
doesn't Action.create return the action method regardless of whether it was successfully created in the database? ([docs](http://api.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-create)) so it seems like this conditional would always be true